qdrant-landing/content/documentation/guides/security.md

```diff
@@ -132,6 +130,209 @@ export QDRANT__SERVICE__READ_ONLY_API_KEY=your_secret_read_only_api_key_here
+
+  ```json
+  {
+    "exp": 1640995200, // Expiration time
+    "value_exists": ..., // Validate this token by looking for a point with a payload value
+    "access": "r", // Define the access level.
+  }
+  ```
+
+- **JWT Signature** - The signature is generated by encoding the header and payload with the `api_key` using the `HS256` algorithm.
+
+  ```
+  HMACSHA256(
+    base64UrlEncode(header) + "." +
+    base64UrlEncode(payload),
+    api_key
+  )
+  ```
+
+#### JWT Payload Claims
+
+- **`exp`** - The expiration time of the token. This is a Unix timestamp in seconds. The token will be invalid after this time.
+
+  ```json
+  {
+    "exp": 1640995200, // Expiration time
+  }
+  ```
+
+- **`value_exists`** - This is a claim that can be used to validate the token at the cost of an extra lookup.
+  
+  ```json
+  {
+    "value_exists": {
+      "collection": "my_validation_collection",
+      "matches": [
+        { "key": "my_key", "value": "value_that_must_exist" }
+      ],
+    },
+  }
+  ```
+
+  If this claim is present, Qdrant will check if there is a point in the collection with the specified key-values. If it does, the token is valid.
+  
+  This is useful when you need stateful tokens which can be invalidated independently.
+
+- **`access`** - This claim defines the [access level](#table-of-access) of the token. If this claim is present, Qdrant will check if the token has the required access level to perform the operation. If this claim is **not** present, **manage** access is assumed.
+
+  It can provide global access with `r` for read-only, or `m` for manage. For example:
+
+  ```json
+  {
+    "access": "r"
+  }
+  ```
+
+  It can also be specific to one or more collections. The `access` level for each collection is `r` for read-only, or `rw` for read-write, like this:
+
+  ```json
+  {
+    "access": [
+      {
+        "collection": "my_collection",
+        "access": "rw"
+      }
+    ]
+  }
+  ```
+
+  You can also specify which subset of the collection the user is able to access by specifying a `payload` restriction that the points must have.
+
+  ```json
+  {
+    "access": [
+      {
+        "collection": "my_collection",
+        "access": "r",
+        "payload": {
+          "user_id": "user_123456"
+        }
+      }
+    ]
+  }
+  ```
+
+  This `payload` claim will be used to implicitly filter the points in the collection, so that they only see their user, for example.
+
+<!-- TODO: add access table -->
+
+### Table of access
+
+Check out this table to see which actions are allowed or denied based on the access level.
+
+This is also applicable to using api keys instead of tokens. In that case, `api_key` maps to **manage**, while `read_only_api_key` maps to **read-only**.
+
+<div style="text-align: right"> <strong>Symbols:</strong> ✅ Allowed | ❌ Denied | 🎟️ Allowed, but filtered </div>
+
+
+
+| Action | manage | read-only | collection read-write | collection read-only | collection with payload claim (r / rw) |
```

Comment:
Is this table auto-generated from the `auth_tests`? If so, we need to document how to regenerate it.

And preferably to get it into the automation, so it wouldn't got outdated when we change rules.

Reply:
No, I wrote it by hand. While automating it is a good idea, everything else in the docs is also manually maintained. Do you have a strong opinion on automating it?

Reply:
It is already present in a machine-readable form, so I think it would be a low-hanging fruit.

Maybe we could put the generation scripts into our repo in the same vein as autogenerated gRPC docs.

Reply:
Any reason people would prefer this table over just trying for themselves? I propose to minimize it to the groups of the operations, which should allow us to e.g. improve on filtered updates without breaking the documented interface.
Sometimes less explicit is better

Reply:
@generall Are you talking about the autogenerated one?